third_party/2and3/werkzeug/formparser.pyi

```diff
@@ -1,45 +1,87 @@
+    silent: bool
+    def __init__(
+        self,
+        stream_factory: Optional[_StreamFactory] = ...,
+        charset: Text = ...,
+        errors: Text = ...,
+        max_form_memory_size: Optional[int] = ...,
+        max_content_length: Optional[int] = ...,
+        cls: Optional[Callable[[], _Dict]] = ...,
+        silent: bool = ...,
+    ): ...
+    def get_parse_func(self, mimetype: str, options: Any) -> Optional[_ParseFunc]: ...
+    def parse_from_environ(self, environ: WSGIEnvironment) -> Tuple[IO[bytes], _Dict, Dict]: ...
+    def parse(
+        self, stream: IO[bytes], mimetype: Text, content_length: Optional[int], options: Optional[Mapping[str, str]] = ...,
+    ) -> Tuple[IO[bytes], _Dict, _Dict]: ...
+    parse_functions: Dict[Text, _ParseFunc]
+
+def is_valid_multipart_boundary(boundary: str) -> bool: ...
+def parse_multipart_headers(iterable: Iterable[Union[Text, bytes]]) -> Headers: ...
+
+class MultiPartParser(object):
     charset: Text
     errors: Text
-    max_form_memory_size: Any
-    stream_factory: Any
-    cls: Any
-    buffer_size: Any
-    def __init__(self, stream_factory: Optional[Any] = ..., charset: Text = ..., errors: Text = ...,
-                 max_form_memory_size: Optional[Any] = ..., cls: Optional[Any] = ..., buffer_size=...): ...
-    def fail(self, message): ...
-    def get_part_encoding(self, headers): ...
-    def get_part_charset(self, headers) -> Text: ...
-    def start_file_streaming(self, filename, headers, total_content_length): ...
-    def in_memory_threshold_reached(self, bytes): ...
-    def validate_boundary(self, boundary): ...
-    def parse_lines(self, file, boundary, content_length, cap_at_buffer: bool = ...): ...
-    def parse_parts(self, file, boundary, content_length): ...
-    def parse(self, file, boundary, content_length): ...
+    max_form_memory_size: Optional[int]
+    stream_factory: _StreamFactory
+    cls: Callable[[], _Dict]
+    buffer_size: int
+    def __init__(
+        self,
+        stream_factory: Optional[_StreamFactory] = ...,
+        charset: Text = ...,
+        errors: Text = ...,
+        max_form_memory_size: Optional[int] = ...,
+        cls: Optional[Callable[[], _Dict]] = ...,
+        buffer_size: int = ...,
+    ) -> None: ...
+    def fail(self, message: Text) -> NoReturn: ...
+    def get_part_encoding(self, headers: Mapping[str, str]) -> Optional[str]: ...
+    def get_part_charset(self, headers: Mapping[str, str]) -> Text: ...
+    def start_file_streaming(
+        self, filename: Union[Text, bytes], headers: Mapping[str, str], total_content_length: Optional[int],
+    ) -> Tuple[Text, IO[bytes]]: ...
+    def in_memory_threshold_reached(self, bytes: Any) -> NoReturn: ...
+    def validate_boundary(self, boundary: Optional[str]) -> None: ...
+    def parse_lines(
+        self, file: Any, boundary: bytes, content_length: int, cap_at_buffer: bool = ...,
+    ) -> Generator[Tuple[str, Optional[bytes]], None, None]: ...
```

Comment:
It also yields some kind of nested tuple: `yield _begin_form, (headers, name)` in https://github.com/pallets/werkzeug/blob/master/src/werkzeug/formparser.py#L471

Reply:
Good point. I just changed the second field type to `Any`.